### PR TITLE
Generate Windows bindings for required symbols only

### PR DESCRIPTION
**Pull request by djc**
This avoids pulling in all of windows-sys and windows (even if most parts are excluded via Cargo features). The windows crate contains 9+ MB of code, so IMO avoiding that makes sense for a low-level/small crate like this.

### CODE REVIEW

**Comment by larseggert** [comment]
Should we exclude the bindings file from coverage checking?

**Comment by djc** [comment]
> Should we exclude the bindings file from coverage checking?

Yes, that probably makes sense. How do you want to do this?

File: src/win_bindings.rs [hunk 1]
```diff
@@ -0,0 +1,452 @@
+// Bindings generated by `windows-bindgen` 0.58.0
```
Comment (martinthomson) [inline src/win_bindings.rs:1]: I have an allergy to checking in generated code.  build.rs?

Comment (djc) [reply src/win_bindings.rs:1]: Why? `build.rs` means all your downstreams have to build an extra compilation unit and extra dependencies just to regenerate some code that ~never changes, and your CI guards against changes in the generated code.

**Comment by larseggert** [comment]
> > Should we exclude the bindings file from coverage checking?
> 
> Yes, that probably makes sense. How do you want to do this?

Change `.codecov.yml` as part of the PR.

File: src/win_bindings.rs (hunk 1 above)
Comment (martinthomson) [reply src/win_bindings.rs:1]: Well, at a minimum, this needs some predefined process for updating.  That might be past of a GitHub action, but checking in generated code without any means to ensure that it continues to track something risks picking up and carrying vulnerabilities.

Comment (djc) [reply src/win_bindings.rs:1]: Maybe a scheduled Actions run?

Comment (martinthomson) [reply src/win_bindings.rs:1]: That's what I was thinking.  Scripting the update is good, running that script periodically is also good, particularly when you can activate all the usual dependency checkers (dependabot etc...).

Comment (djc) [reply src/win_bindings.rs:1]: Added a separate commit with this.

Comment (larseggert) [reply src/win_bindings.rs:1]: So the intention here is that whenever CI fails, we go and recreate an updated version of the bindings and then commit them?

Comment (djc) [reply src/win_bindings.rs:1]: Yes, that's the idea. From my experience with other projects using this approach (in particular, chrono), this almost never happens.

**Review by larseggert — APPROVED** [review]

**Comment by larseggert** [comment]
Thanks!

**Pull request merged**